lib/Controller/ConsentAdmin.php

```diff
@@ -0,0 +1,418 @@
+     */
+    public function setMetadataStorageHandler(MetadataStorageHandler $handler): void
+    {
+        $this->metadataStorageHandler = $handler;
+    }
+
+
+    /**
+     * Inject the \SimpleSAML\Module\consent\Auth\Process\Consent dependency.
+     *
+     * @param \SimpleSAML\Module\consent\Auth\Process\Consent $consent
+     */
+    public function setConsent(Consent $consent): void
+    {
+        $this->consent = $consent;
+    }
+
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request The current request.
+     *
+     * @return \SimpleSAML\XHTML\Template
+     */
+    public function main(Request $request): Template
+    {
+        $authority = $this->moduleConfig->getValue('authority');
+
+        $as = new $this->authSimple($authority);
+
+        // If request is a logout request
+        $logout = $request->get('logout');
+        if ($logout !== null) {
+            $returnURL = $this->moduleConfig->getValue('returnURL');
+            $as->logout($returnURL);
+        }
+
+        $hashAttributes = $this->moduleConfig->getValue('attributes.hash', false);
+
+        $excludeAttributes = $this->moduleConfig->getValue('attributes.exclude', []);
+
+        // Check if valid local session exists
+        $as->requireAuth();
+
+        // Get released attributes
+        $attributes = $as->getAttributes();
+
+        // Get metadata storage handler
+        $metadata = $this->metadataStorageHandler;
+
+        /*
+         * Get IdP id and metadata
+         */
+        $idp_entityid = $metadata->getMetaDataCurrentEntityID('saml20-idp-hosted');
+        $idp_metadata = $metadata->getMetaData($idp_entityid, 'saml20-idp-hosted');
+
+        // Calc correct source
+        if ($as->getAuthData('saml:sp:IdP') !== null) {
+            // from a remote idp (as bridge)
+            $source = 'saml20-idp-remote|' . $as->getAuthData('saml:sp:IdP');
+        } else {
+            // from the local idp
+            $source = $idp_metadata['metadata-set'] . '|' . $idp_entityid;
+        }
+
+        // Get user ID
+        if (isset($idp_metadata['userid.attribute']) && is_string($idp_metadata['userid.attribute'])) {
+            $userid_attributename = $idp_metadata['userid.attribute'];
+        } else {
+            $userid_attributename = 'eduPersonPrincipalName';
+        }
+
+        $userids = $attributes[$userid_attributename];
+
+        if (empty($userids)) {
+            throw new Exception(sprintf(
+                'Could not generate useridentifier for storing consent. Attribute [%s] was not available.',
+                $userid_attributename
+            ));
+        }
+
+        $userid = $userids[0];
+
+        // Get all SP metadata
+        $all_sp_metadata = $metadata->getList('saml20-sp-remote');
+
+        $sp_entityid = $request->get('cv');;
+        $action = $request->get('action');
+
+        Logger::critical('consentAdmin: sp: ' . $sp_entityid . ' action: ' . $action);
```

Comment:
This unconditionally logs a critical? Why?

Reply:
Good question, was a copy-paste..   I'll reduce it to notice